jvmquake.c

```diff
@@ -155,6 +165,57 @@ gc_finished(jvmtiEnv *jvmti) {
             err = (*jvmti)->RawMonitorExit(jvmti, lock);
             error_check(err, "Failed to notify killer thread");
         }
+    } else if (val > opt_gc_warning_threshold && opt_gc_warning_path != NULL) {
+        fprintf(stderr,
+                "(jvmquake) Above GC warning threshold [%lds]: touching (%s)\n",
+                opt_gc_warning_threshold / NANOS, opt_gc_warning_path);
+        int fd = open(opt_gc_warning_path, O_CREAT | O_WRONLY, 0666);
+        if (fd >= 0) {
+            futimes(fd, NULL);
+        } else {
+            fprintf(stderr, "(jvmquake) ERROR: Could not touch (%s), error (%d)\n",
+                    opt_gc_warning_path, errno);
+        }
+    }
+}
+
+static void
+parse_kwargs(char *kwargs) {
+    if (strlen(kwargs) == 0) { return; }
+    const char comma[2] = ",";
+
+    char * savestate;
+    char * key;
+    char * value;
+    char * equal_pos;
+
+    key = strtok_r(kwargs, comma, &savestate);
+
+    while (key != NULL) {
+        equal_pos = strchr(key, '=');
+        if (equal_pos != NULL) {
+            // Now we have ... two valid strings
+            *equal_pos = '\0';
+            value = equal_pos + 1;
+            if (!strncmp(key, KWARG_WARN, strlen(KWARG_WARN))) {
+                opt_gc_warning_threshold = atol(value) * NANOS;
+            } else if (!strncmp(key, KWARG_TOUCH, strlen(KWARG_TOUCH))) {
+                strncpy(opt_gc_warning_path, value, 255);
+            }
+        } else {
+            fprintf(stderr, "(jvmquake): WARN: no equals in kwarg");
+        }
+        key = strtok_r(NULL, comma, &savestate);
+    }
+
+    if (opt_gc_warning_threshold < ULONG_MAX) {
+        if (opt_gc_warning_path[0] == '\0') {
+            fprintf(stderr, "(jvmquake): ERROR: Cannot set warn without touch\n");
+            abort();
+        }
+        fprintf(stderr,
+                "(jvmquake) using kwarg options: warn_threshold=[%lds],touch_path=[%s]\n",
```

Comment:
Continuing with the idea of "not everyone knows what a kwarg is", could we say "keyword arguments"?

Reply:
done.